### PR TITLE
[testutils] Simplify count down logic in HubAccumulator

### PR DESCRIPTION
**Pull request by jamesonwilliams**
By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java [hunk 1]
```diff
@@ -139,28 +144,20 @@ public HubAccumulator clear() {
...
-    public List<HubEvent<?>> take(int desiredQuantity) {
-        // If we haven't yet received the desired quantity of events on the subscription,
-        // setup a latch to await the desired quantity, less the number of existing events.
-        // For example: I desire 5, I already have 3, I wait for 2 more.
-        if (events.size() < desiredQuantity) {
-            latch = new CountDownLatch(desiredQuantity - events.size());
-            Latch.await(latch);
-            latch = null;
-        }
+    public List<HubEvent<?>> takeAll() {
```
Comment (raphkim) [inline testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:152]: is there any reason to specify "all" now that it's the only option you provide? I would suggest just `take` but that doesn't sound like its very descriptive of what it's doing. What would be a good verb to follow up on accumulating something? `drain`? Maybe `consume`?

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java [hunk 2]
```diff
@@ -139,28 +144,20 @@ public HubAccumulator clear() {
...
+        Latch.await(latch);
 
         // If we already had the right number of events,
         // or if our latch counted down as a result of receiving the remaining number of events,
         // return the requested number of events.
```
Comment (raphkim) [inline testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:157]: is this comment still relevant?

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java [hunk 3]
```diff
@@ -139,28 +144,20 @@ public HubAccumulator clear() {
      * block until the rest show up. If there are enough, they will
      * be returned immediately. The returned items will be cleared from the
      * accumulator.
-     * @param desiredQuantity Number of items being awaited
      * @return A list of desiredQuantity many items
```
Comment (raphkim) [inline testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:147]: `desiredQuantity` is no longer a parameter

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java [hunk 4]
```diff
@@ -139,28 +144,20 @@ public HubAccumulator clear() {
...
-            latch = null;
-        }
+    public List<HubEvent<?>> takeAll() {
+        Latch.await(latch);
 
         // If we already had the right number of events,
         // or if our latch counted down as a result of receiving the remaining number of events,
         // return the requested number of events.
-        List<HubEvent<?>> returning = new ArrayList<>(events.subList(0, desiredQuantity));
+        CopyOnWriteArrayList<HubEvent<?>> returning = new CopyOnWriteArrayList<>(events);
```
Comment (raphkim) [inline testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:158]: Whoa any reason to do this? Will a user ever need to write anything on the returned list? Can't you just return an immutable list of hub events?

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/AWSDataStorePluginInstrumentedTest.java [hunk 5]
```diff
@@ -181,7 +181,7 @@ public void blogOwnerCreatedAndUpdatedRemotelyIsFoundLocally() throws DataStoreE
 
         // Another HubEvent tells us that an update occurred in the Cloud;
         // the update was applied locally, to an existing record.
-        inboundModelEventAccumulator.take(1);
+        inboundModelEventAccumulator.takeAll();
```
Comment (raphkim) [inline aws-datastore/src/androidTest/java/com/amplifyframework/datastore/AWSDataStorePluginInstrumentedTest.java:184]: Isn't `HubAccumulator` now not-recyclable after this code change? Wouldn't calling `takeAll()` the second time break it?

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java (hunk 1 above)
Comment (jamesonwilliams) [reply testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:152]: I like `drain`. I checked Merriam Webster for antonyms of "accumulate", but none of them felt appropriate here. The "all" was added to indicate the collection return type.

`drain` actually describes the behavior a little better though. This method does drain all items out of the accumulator.

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java (hunk 2 above)
Comment (jamesonwilliams) [reply testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:157]: NO! Thanks, good catch.

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java (hunk 3 above)
Comment (jamesonwilliams) [reply testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:147]: Thanks, will fix

File: testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java (hunk 4 above)
Comment (jamesonwilliams) [reply testutils/src/main/java/com/amplifyframework/testutils/HubAccumulator.java:158]: Good point!

File: aws-datastore/src/androidTest/java/com/amplifyframework/datastore/AWSDataStorePluginInstrumentedTest.java (hunk 5 above)
Comment (jamesonwilliams) [reply aws-datastore/src/androidTest/java/com/amplifyframework/datastore/AWSDataStorePluginInstrumentedTest.java:184]: You're right; it is no longer recycle-able. I would expect that calling `takeAll()` the second time would return an empty list immediately.

**Comment by jamesonwilliams** [comment]
This PR has gone stale. Let's re-open it, if/when it will receive an update. It is still here for reference, should we wish to use it for that purpose, meanwhile.

**Pull request closed without merging**